Framework/Algorithms/src/GeneratePythonFitScript.cpp

```diff
@@ -224,6 +246,64 @@ std::string GeneratePythonFitScript::generateVariableSetupCode() const {
   return code;
 }
 
+std::string GeneratePythonFitScript::generateSequentialFitCode() const {
+  std::string code = "# Perform a sequential fit\n";
+  code += "output_workspaces, parameter_tables, normalised_matrices = [], [], []\n";
+  code += "for input_workspace, domain_data in input_data.items():\n";
+  code += "    fit_output = Fit(Function=function, InputWorkspace=input_workspace, WorkspaceIndex=domain_data[0],\n";
+  code += "                     StartX=domain_data[1], EndX=domain_data[2], MaxIterations=max_iterations,\n";
+  code += "                     Minimizer=minimizer, CostFunction=cost_function, EvaluationType=evaluation_type,\n";
+  code += "                     CreateOutput=True)\n";
+  code += "\n";
+  code += "    output_workspaces.append(fit_output.OutputWorkspace)\n";
+  code += "    parameter_tables.append(fit_output.OutputParameters)\n";
+  code += "    normalised_matrices.append(fit_output.OutputNormalisedCovarianceMatrix)\n";
+  code += "\n";
+  code += "    # Use the parameters in the previous function as the start parameters of the next fit\n";
+  code += "    function = fit_output.Function\n";
+  code += "\n";
+  code += "# Group the output workspaces from the sequential fit\n";
+  code += "GroupWorkspaces(InputWorkspaces=output_workspaces, OutputWorkspace=\"Sequential_Fit_Workspaces\")\n";
+  code += "GroupWorkspaces(InputWorkspaces=parameter_tables, OutputWorkspace=\"Sequential_Fit_Parameters\")\n";
+  code += "GroupWorkspaces(InputWorkspaces=normalised_matrices, ";
+  code += "OutputWorkspace=\"Sequential_Fit_NormalisedCovarianceMatrices\")\n\n";
+  return code;
```

Comment:
I'm not keen on leaving the template as a string in C++, as it becomes much harder to maintain.

Is there a way we can save this into a dedicated .py file alongside this algorithm, then simply load it in (and do template substitution if required in the future)? 
That way we can still run static analysis like PEP8, and use dedicated Python tooling to edit the files.

Reply:
Do you know of any examples of this in Mantid so I can see how this might be done?

Reply:
What location do you think I should save the template python files?

Reply:
> Do you know of any examples of this in Mantid so I can see how this might be done?
Unfortunately not, as we tend to steer people away from emitting Python from C++, as they're usually doing the wrong thing. This is the exception to the rule, but any file loading algorithm will give 80% of the code (as loading is the hardest part) plus the snippet I gave below


> What location do you think I should save the template python files?
I would save it alongside the *.cpp with the name of the algorithm in the name. If someone feels like there is a better place to place them I'm happy to take suggestions. But I'd rather just get a *.py file in the first place

Reply:
Further thinking about it, the hardest part will be building a file path to this template. Specifically getting the install directory of Mantid.
I'm trying to find an example in the codebase where someone has already done that (e.g. for instrument definitions we ship)

Reply:
The template python files get caught by pre-commit because they are not completely formatted as expected due to the `{{insert_here}}` dotted around in the file. Also I have split up the templates into several files so some files are missing imports (which has been imported by a previous template file)

@DavidFair Maybe I should use `*.txt` files instead of `*.py` files because they are not properly formatted python files? Or would you suggest a different file format?

Reply:
I'm happy with *.txt it's obvious it's Python code and someone can easily rename it to *.py if they want their tooling to work so badly